fix: ch3 automining

rin-st opened the pull request:
Fixes https://github.com/scaffold-eth/se-2-challenges/issues/243

rin-st commented:
note: currently our alchemy key doesnt work, to test a challenge you need to changeDEFAULT_ALCHEMY_API_KEY in scaffold.config for example to `7Y0hZLkHFZMDyCj5k1n2y3TE1FTQM5Fo`

damianmarti reviewed (approved):
Thanks @rin-st!

Working ok and the code looks good!!

We should add an issue to remove this manual mining when we add the option to change hardhat config to automining from create-eth.

rin-st commented:
Updated code a bit: removed public client from dice pate [034ee0d](https://github.com/scaffold-eth/se-2-challenges/pull/253/commits/034ee0d9bb3082eb1fae9aee3e189cc6a191234f)

Created an issue to remove mining when possible #255

Thanks! Merging this
